component/file_cache/file_cache.go

```diff
@@ -843,7 +843,7 @@ func (fc *FileCache) downloadFile(handle *handlemap.Handle) error {
 		}
 
 		// Open the file in write mode.
-		f, err = common.OpenFile(localPath, os.O_CREATE|os.O_RDWR, fMode)
+		f, err = common.OpenFile(localPath, flags, fMode)
```

Comment:
The flags we use here are for downloading object data from the cloud. We should not use append for that, right?

Reply:
Maybe a "case 1" unit test would have caught this. Do you want to add one?

Reply:
Good catch. Actually, this was not needed to fix O_APPEND and accidently included while I was trying to track the permissions issue down. So I have reverted this change.